src/components/cart/CheckRecipe.tsx

```diff
@@ -0,0 +1,136 @@
+import { Box, Flex, Image, Text } from "@chakra-ui/react";
+import { AddIcon, MinusIcon, SmallCloseIcon } from "@chakra-ui/icons";
+import { FC } from "react";
+import { useMyStore } from "../store/store.tsx";
+import { NewRecipe, Product } from "../types.ts";
+import { useGetRecipePrice } from "../hooks/useGetRecipePrice.tsx";
+
+type Props = {
+  recipeInCart: string;
+};
+
+type IngredientData = null | {
+  productsByStoreId: {
+    [storeId: string]: Product[];
+  };
+  ingredientIds: {
+    [storeId: string]: string[];
+  };
+};
+
+const CheckRecipe: FC<Props> = ({ recipeInCart }) => {
+  const { recipes } = useMyStore();
+  const findRecipeById = (recipeInCart: string) =>
+    recipes.find((oneRecipe) => oneRecipe.id === recipeInCart);
+
+  // @ts-ignore
+  const specificRecipe: NewRecipe = findRecipeById(recipeInCart);
+  const { productIds, ingredientData } = useGetRecipePrice(specificRecipe);
+
+  const getFilteredIngredientData = (
```

Comment:
Tohle je hezka [pure function](https://en.wikipedia.org/wiki/Pure_function). Pokud se dobre koukam, neni nutne duvod aby byla ve stejne scope jako ta komponenta. Klidne ji muzem vytahnout ven a pak pro ni napsat test. Take bych se vyvaroval tomu, ze `ingredientData` muzou chybet, byl bych tam trochu duraznejsi a rekl, ze ne (to, ze ne vzdy mas ingredientData nemusi byt nutne problem tenhle funkce). Pak se ti to i zjednodussi. Co ty na to?